source/includes/connection-snippets/x509-snippets/csharp-connection.cs

```diff
@@ -0,0 +1,53 @@
+// begin x509 connection
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Net.Security;
+
+using MongoDB.Bson;
+using MongoDB.Driver;
+using MongoDB.Driver.Core;
+using System;
+using System.Threading.Tasks;
+
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Cryptography;
+using System.Security.Permissions;
+
+namespace WorkingWithMongoDB
+{
+    public class Entity
+    {
+        public ObjectId Id { get; set; }
+        public string Name { get; set; }
+    }
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            MainAsync().Wait();
+            Console.WriteLine("done");
+        }
+
+        static async Task MainAsync()
+        {
+            var settingObjectOnlySettings = new MongoClientSettings 
```

Comment:
Probably should rename this to something like "settings"

Reply:
Fixed